adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/OAuthRequestAuthenticator.java

```diff
@@ -375,6 +381,23 @@ protected String stripOauthParametersFromRedirect() {
                 .replaceQueryParam(OAuth2Constants.STATE, null);
         return builder.build().toString();
     }
-
+    
+    private String rewrittenRedirectUri(String originalUri) {
+        String rewrittenRedirectUri = originalUri;
+        try {
+            URL url = new URL(originalUri);
+            Map<String, String> rewriteRules = deployment.getRedirectRewriteRules();
+            if(rewriteRules != null && !rewriteRules.isEmpty()) {
+                Map.Entry<String, String> rule =  rewriteRules.entrySet().iterator().next();
+                StringBuilder redirectUriBuilder = new StringBuilder(url.getProtocol());
+                redirectUriBuilder.append("://"+ url.getAuthority());
+                redirectUriBuilder.append(url.getPath().replaceFirst(rule.getKey(), rule.getValue()));
+                return redirectUriBuilder.toString();
+            }
+        } catch (MalformedURLException ex) {
+            log.error("Not a valid request url");
```

Comment:
Should it not throw an error instead of continuing with what is most likely the wrong url?